### DISCUSSION TITLE
Netbox Secret Store - Server Error

### DISCUSSION THREAD

**Discussion by lill-peter** [netbox-community/netbox#12282] [Help Wanted!]
Hi,
I get a server error when i try to add new Secret to a Device. I can see secret on old device / secret.

Server Error
There was a problem with your request. Please contact an administrator.
The complete exception is provided below:
<class 'netbox_secretstore.models.secrets.Secret.assigned_object_type.RelatedObjectDoesNotExist'>
Secret has no assigned_object_type.
Python version: 3.8.10
NetBox version: 3.4.8

/var/log/nginx/access.log
::ffff:x.x.x.x - - [19/Apr/2023:09:12:24 +0200] "POST /plugins/netbox_secretstore/secrets/add/?device=335&return_url=/dcim/devices/335/ HTTP/1.1" 500 1658 "https://netbox.domain.com/plugins/netbox_secretstore/secrets/add/?device=335&return_url=/dcim/devices/335/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"

::ffff:x.x.x.x - - [19/Apr/2023:09:12:24 +0200] "GET /static/netbox-light.css HTTP/1.1" 200 232647 "https://netbox.domain.com/plugins/netbox_secretstore/secrets/add/?device=335&return_url=/dcim/devices/335/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"

**Comment by DanSheps** [comment]
It is recommended to transition to netbox-secrets instead of netbox-secretstore.

As far as this error, likely you are missing a migration or something similar

**Reply by lill-peter** [reply]
Thanks @DanSheps for the help. I will migrate to netbox-secrets.

**Reply by DanSheps** [reply]
If your FK relations are missing, there is likely other issues that need to be addressed, FYI.
